src/Language/Printer.php

```diff
@@ -181,12 +181,12 @@ public function printAST($ast)
                             . $node->selectionSet;
                     },
 
-                    NodeKind::INT => static function (IntValueNode $node) {
-                        return $node->value;
+                    NodeKind::INT => static function (IntValueNode $node) : string {
+                        return (string) $node->value;
```

Comment:
`$node->value` must always be a string. So no need to cast in this case - the error is actually here:

https://github.com/webonyx/graphql-php/blob/2dcc81e4d7f01470e66c72a2a1db8050c8b0b70a/src/Utils/AST.php#L254-L268

Instead, we must cast `$serialized` to string when creating all of those nodes.